tech-far-hub/src/pages/get-started/template-default.tsx

```diff
@@ -2,13 +2,13 @@ import * as React from "react";
 import type { HeadFC, PageProps } from "gatsby";
 import { graphql } from "gatsby";
 import Layout from "../../components/layout";
+import { IPageContext } from "../../types";
 
-const LifecyclePage: React.FC<PageProps<Queries.GetStartedInnerPageQuery>> = ({
-  data,
-  children,
-}: PageProps<Queries.GetStartedInnerPageQuery>) => {
+type LifecyclePageProps = PageProps<Queries.GetStartedInnerPageQuery, IPageContext>;
```

Comment:
One of these you'll have to explain this to me! Nice work!

Reply:
Gladly. The short version is PageProps is a generic that can be specified across a couple  parameters. So in this case, I'm passing both the query type and the stuff i expect to get in the PageContext. It also takes a custom type for LocationState and ServerData, but I don't pass those in yet (maybe never tbh).